beeline-core/src/main/java/io/honeycomb/beeline/tracing/propagation/HttpClientPropagator.java

```diff
@@ -50,24 +53,18 @@
      * @param requestToSpanName a function from request to span name
      */
     public HttpClientPropagator(final Tracer tracer, final Function<HttpClientRequestAdapter, String> requestToSpanName) {
-        this(tracer, Propagation.honeycombHeaderV1(), requestToSpanName);
+        this(tracer, Propagation.honeycombHeaderV1(), requestToSpanName, null);
     }
 
-    /**
-     * Create an HttpClientPropagator for tracing HTTP client requests.
-     * {@code requestToSpanName} allows you to dynamically name the HTTP client spans such that the name
-     * reflects the operation, e.g. based on HTTP method or request path used.
-     *
-     * @param tracer the tracer
-     * @param propagationCodec the propagation codec to use for parsing and propagating trace data
-     * @param requestToSpanName a function from request to span name
-     */
-    public HttpClientPropagator(final Tracer tracer,
+    // User by builder
+    protected HttpClientPropagator(final Tracer tracer,
                                 final PropagationCodec<Map<String, String>> propagationCodec,
-                                final Function<HttpClientRequestAdapter, String> requestToSpanName) {
+                                final Function<HttpClientRequestAdapter, String> requestToSpanName,
+                                final BiFunction<HttpClientRequestAdapter, PropagationContext, Optional<Map<String, String>>> tracePropagationHook) {
```

Comment:
TIL about Function types in Java 8 :-) Nice!

Reply:
Yeah, they are pretty useful for creating in-line lambda functions. Still took me some time to figure out how they word coming from .NET to Java 😉